chips/imxrt1050/src/lpi2c.rs

```diff
@@ -0,0 +1,807 @@
+
+        self.registers.mcfgr1.modify(MCFGR1::PINCFG::CLEAR);
+
+        self.registers.mier.modify(
+            MIER::TDIE::SET + MIER::NDIE::SET + MIER::RDIE::SET + MIER::EPIE::SET + MIER::FEIE::SET,
+        );
+    }
+
+    fn stop(&self) {
+        self.tx_position.set(0);
+        self.tx_len.set(0);
+        self.rx_position.set(0);
+        self.rx_len.set(0);
+        self.status.set(Lpi2cStatus::Idle);
+        if self.registers.msr.is_set(MSR::FEF) {
+            self.registers.msr.modify(MSR::FEF::SET);
+        }
+        self.registers.mier.modify(
+            MIER::TDIE::CLEAR
+                + MIER::NDIE::CLEAR
+                + MIER::EPIE::CLEAR
+                + MIER::SDIE::CLEAR
+                + MIER::RDIE::CLEAR,
+        );
+    }
+
+    fn start_read(&self) {
+        self.rx_position.set(0);
+
+        // setting slave address
+        self.registers.mier.modify(MIER::EPIE::CLEAR);
+        self.registers
+            .mtdr
+            .write(MTDR::CMD.val(100) + MTDR::DATA.val((self.slave_address.get() << 1 + 1) as u32));
+
+        self.registers.mcfgr1.modify(MCFGR1::PINCFG::CLEAR);
+        self.registers
+            .mier
+            .modify(MIER::NDIE::SET + MIER::EPIE::SET + MIER::RDIE::SET);
+    }
+}
+
+impl i2c::I2CMaster for Lpi2c<'_> {
+    fn set_master_client(&self, master_client: &'static dyn I2CHwMasterClient) {
+        self.master_client.replace(master_client);
+    }
+    fn enable(&self) {
+        self.registers.mcr.modify(MCR::RRF::SET);
+        self.registers.mcr.modify(MCR::RTF::SET);
+        self.registers.mcr.modify(MCR::MEN::SET);
+    }
+    fn disable(&self) {
+        self.registers.mcr.modify(MCR::MEN::CLEAR);
+    }
+    fn write_read(&self, addr: u8, data: &'static mut [u8], write_len: u8, read_len: u8) {
+        debug!("write_read {:?} {} {}", data, write_len, read_len);
+        if self.status.get() == Lpi2cStatus::Idle {
+            self.reset();
+            self.status.set(Lpi2cStatus::WritingReading);
+            self.slave_address.set(addr);
+            self.buffer.replace(data);
+            self.tx_len.set(write_len);
+            self.rx_len.set(read_len);
+            self.registers.mcfgr1.modify(MCFGR1::AUTOSTOP::CLEAR);
+            debug!("Inainte de start write  ");
+            self.start_write();
+        }
+    }
+
+    fn write(&self, addr: u8, data: &'static mut [u8], len: u8) {
+        debug!("write {:?} {}", data, len);
+        if self.status.get() == Lpi2cStatus::Idle {
+            self.reset();
+            self.status.set(Lpi2cStatus::Writing);
+            self.slave_address.set(addr);
+            self.buffer.replace(data);
+            self.tx_len.set(len);
+            self.registers.mcfgr1.modify(MCFGR1::AUTOSTOP::CLEAR);
+            self.start_write();
+        }
+    }
+
+    fn read(&self, addr: u8, buffer: &'static mut [u8], len: u8) {
+        debug!("read {:?} {}", buffer, len);
```

Comment:
I suspect these `debug!()` statements be removed before merging

Reply:
Deleted them.